/*
 * Copyright the original author or authors.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */

package de.schildbach.oeffi;

import java.util.List;

import de.schildbach.pte.dto.Point;
import de.schildbach.pte.dto.Trip;
import de.schildbach.pte.dto.Trip.Leg;

public interface TripAware {
    interface LegInfo {
        boolean isPublicLeg();
        Leg getLeg();
        List<Point> getPath();
    }

    Trip getTrip();

    int getNumberOfLegs();

    LegInfo getLegInfo(int legIndex);

    void selectLeg(int legIndex);

    boolean hasSelection();

    boolean isSelectedLeg(int legIndex);
}
